### ISSUE TITLE
Minor Error on focus-order-semantics with tabindex 0 on role tooltip element

### ISSUE THREAD

**Issue by Sly321** [dequelabs/axe-core#2813]
Expectation: With these information from the [documentation](https://dequeuniversity.com/rules/axe/4.1/focus-order-semantics) that an element with the widget role `tooltip` is an appropriate element for a `tabindex=0` declaration.

Actual: Axe Core 4.1.2 giving me this failure: `Fix any of the following:↵  Element does not have a widget role.↵  Element has invalid semantics for an element in the focus order.` on this element:

```html
<span id="4dow0v" role="tooltip" tabindex="0" />
```

the hole implementation looks like this (cutted style and classNames):

```html
<span aria-describedby="gju2b0">
    <span id="gju2b0" role="tooltip" tabindex="0">
        <span>The element is inactive because it wasn't adressed for 1 week</span>
    </span>
    <span>Inactive</span>
</span>
```

I get these errors when executing the following rules:

```javascript
runOnly: {
    type: "rule",
    values: ["focus-order-semantics", "tabindex", "skip-link", "scrollable-region-focusable"],
}
```

An example for the hole Error description that i get from axe:

```json
{
  "any": [
    {
      "id": "has-widget-role",
      "data": null,
      "relatedNodes": [],
      "impact": "minor",
      "message": "Element does not have a widget role."
    },
    {
      "id": "valid-scrollable-semantics",
      "data": null,
      "relatedNodes": [],
      "impact": "minor",
      "message": "Element has invalid semantics for an element in the focus order."
    }
  ],
  "all": [],
  "none": [],
  "impact": "minor",
  "html": "<span id=\"4dow0v\" role=\"tooltip\" class=\"tooltip fade bs-tooltip-top\" tabindex=\"0\" style=\"top: 0px; left: 50%; margin-left: 0px; transform: translate(-50%, -100%); display: inline-block; pointer-events: none; position: fixed;\">",
  "target": [
    "span[title=\"Einrichtungsnummer\"] > .tooltip.bs-tooltip-top[role=\"tooltip\"]"
  ],
  "failureSummary": "Fix any of the following:\n  Element does not have a widget role.\n  Element has invalid semantics for an element in the focus order."
}
```

Motivation: I assume this is an bug, but i am not totally sure, because i do not get this error when i test my page with the browser extension of axe, but the browser extension runs with axe 4.7.1 and i am not sure if this means it is also axe-core 4.7.1. Thanks for the Help! 

<pre><code>
axe-core version: 4.1.2
axe-webdriver, extension or other integration version: cypress-axe@0.12.1

Browser and Assistive Technology versions

For Tooling issues:
- Node version: v14.15.4
- Platform:  Windows 10
</code></pre>

**Comment by straker** [comment]
Thanks for the issue. It appears our documentation is incorrect as `tooltip` is not a widget role. However, we also believe this shouldn't fail and tabindex should be allowed on `tooltip` so we will create an option for the rule that allows `tooltip` to pass.

**Comment by padmavemulapati** [comment]
Verified with the latest axe-core code base, I am unable to reproduce the issue with the specified rule version, and also with the latest axe version. 
Used code snippet `<span id="sample" role="tooltip" tabindex="0">` also 
`<div id="sample2" role="tooltip" tabindex="0"></div>`
-- not seeing any violation  for `valid-scrollable-semantics`

![image](https://user-images.githubusercontent.com/47409579/116096918-4c13c980-a6c7-11eb-95f2-fa27a84a9664.png)
